client/src/partials/debtors/groups.update.js

```diff
@@ -44,15 +44,17 @@ function DebtorGroupsUpdateController($state, DebtorGroups, Accounts, Prices, Sc
     // ensure we don't make HTTP requests if the form is invalid - exit early
     if (debtorGroupForm.$invalid) {
       Notify.danger('FORM.ERRORS.RECORD_ERROR');
-      return false;
+      return;
     }
 
-    /** @todo filterDirtyFormElements should be updated to factor in nested forms */
-    // submitDebtorGroup = util.filterDirtyFormElements(debtorGroupForm);
-    submitDebtorGroup = angular.copy(vm.group);
+    // catch 'nothing has changed' and redirect to list page
+    if (debtorGroupForm.$pristine) {
```

Comment:
Just an idea: If we have enough of these checks, we could abstract them into a shared function that errors for form errors, warns and redirects for `$pristine`, etc.

This looks like a nice improvement for UX!